tests/PHPUnit/ApiClient/Helper/PartnerAttributionTest.php

```diff
@@ -0,0 +1,121 @@
+<?php
+declare(strict_types=1);
+
+namespace WooCommerce\PayPalCommerce\ApiClient\Helper;
+
+use PHPUnit\Framework\TestCase;
+
+use function Brain\Monkey\Functions\when;
+use function Brain\Monkey\Functions\expect;
+use function Brain\Monkey\setUp;
+use function Brain\Monkey\tearDown;
+
+/**
+ * Test case for PartnerAttribution class.
+ */
+class PartnerAttributionTest extends TestCase {
+
+	/**
+	 * Option name for storing the BN Code.
+	 *
+	 * @var string
+	 */
+	private string $bn_code_option_name = 'ppcp_bn_code';
+
+	/**
+	 * BN codes mapping.
+	 *
+	 * @var array
+	 */
+	private array $bn_codes = [
+		'core-profiler'    => 'WooPPCP_Ecom_PS_CoreProfiler',
```

Comment:
I would also use the `ActivationDetector` const values for the array keys, but keep the string values for BN codes, as we must only test if the BN code changes, and do not really care if/when the internal const value changes:

```php
private array $bn_codes = [
  ActivationDetector::CORE_PROFILER    => 'WooPPCP_Ecom_PS_CoreProfiler',
  ActivationDetector::PAYMENT_SETTINGS => 'WooPPCP_Ecom_PS_CoreProfiler',
);
```

Reply:
Already done :)